main.tf

```diff
@@ -39,6 +39,7 @@ module "project-factory" {
   shared_vpc                         = var.svpc_host_project_id
   enable_shared_vpc_service_project  = var.svpc_host_project_id != ""
   enable_shared_vpc_host_project     = var.enable_shared_vpc_host_project
+  grant_services_network_role        = var.grant_services_network_role
```

Comment:
I am inclined to rename this from `grant_services_network_role` to `grant_network_role` since this is no longer limited to service agents.